File: 800-Rating_codeforces/1903A.py
def main():
    t = int(input())
    while t > 0:
        t -= 1
        n, k = map(int, input().split())
        a = list(map(int, input().split()))
        is_sorted = True

        for i in range(1, n):
            if a[i] < a[i - 1]:
                is_sorted = False
                break

        if is_sorted:
            print("YES")
        else:
            if 1 < k <= n:
                print("YES")
            else:
                print("NO")

if __name__ == "__main__":
    main()

File: 800-Rating_codeforces/1903A.cpp
#include<bits/stdc++.h> 
using namespace std ;

int main() {
    int t ;
    cin >> t ;
    while(t-- ) {
        int n , k ;
        cin >> n >> k ; 
        int a[n] ;
        int isSorted = 1 ; 
        int isReverseSorted = 1 ;
        for(int i = 0 ; i < n ; i++ ) {
            if(i == 0 ) {
                cin >> a[i] ;
            } else {
                cin >> a[i] ;
                if(a[i] <  a[i-1] ) {
                    isSorted = 0 ;
                }
            }
        }
        if( isSorted == 1 ) {
            cout << "YES" << endl ; 
        }   else {
            if ( k <= n && k > 1 ) {
                cout << "YES" << endl ; 
                continue ; 
            }
            cout << "NO" << endl ;

        }
    }
    return 0 ; 

}